src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseHelper.CoreClr.cs

```diff
@@ -36,11 +36,34 @@ internal static class WebResponseHelper
         }
 
         internal static string GetOutFilePath(HttpResponseMessage response, string _qualifiedOutFile)
-        {
-            // Get file name from last segment of Uri
-            string lastUriSegment = System.Net.WebUtility.UrlDecode(response.RequestMessage.RequestUri.Segments[^1]);
+        {            
+            if (Directory.Exists(_qualifiedOutFile))
+            {
+                string contentDisposition = response.Content.Headers.ContentDisposition?.FileNameStar ?? response.Content.Headers.ContentDisposition?.FileName;
+                string pathAndQuery = response.RequestMessage.RequestUri.PathAndQuery;
+
+                if (!string.IsNullOrEmpty(contentDisposition)) 
+                {
+                    // Get file name from Content-Disposition header if present
+                    return Path.Join(_qualifiedOutFile, contentDisposition);
+                }
+                else if (pathAndQuery != "/")
```

Comment:
We can remove `else` here and below since `return`.

Reply:
done